### PR TITLE
avoid repeated length calculation and some other code improvements

### PR DESCRIPTION
**Pull request by bruceauyeung**
**What this PR does / why we need it**:
1. in function `ParsePairs`,  calculating `invalidBuf`'s length over and over again brings performance penalty. a `invalidBufNonEmpty` bool value can fix this.
2. pairArg is not a string template and also there is no other arguments for `fmt.Sprintf`, so i remove `fmt.Sprintf`
3. in function `DumpReaderToFile`, we must check nil error first before defer statement, otherwise there maybe a potential nil error on `f.Close()`
4. add nil checks into `GetWideFlag` function
5. some other minor code improvements for better readability.

Signed-off-by: bruceauyeung <ouyang.qinhua@zte.com.cn>

### CODE REVIEW

**Comment by dims** [comment]
@k8s-bot ok to test

**Comment by bruceauyeung** [comment]
@AdoHe 
hi, would you please check if this is a resonable PR, thanks !
cc @smarterclayton @brendandburns

File: pkg/kubectl/cmd/util/helpers.go [hunk 1]
```diff
@@ -591,28 +589,30 @@ func ParsePairs(pairArgs []string, pairType string, supportRemove bool) (newPair
 		removePairs = []string{}
 	}
 	var invalidBuf bytes.Buffer
-
+	var invalidBufNonEmpty bool
 	for _, pairArg := range pairArgs {
-		if strings.Index(pairArg, "=") != -1 {
+		if strings.Contains(pairArg, "=") {
```
Comment (AdoHe) [inline pkg/kubectl/cmd/util/helpers.go]: any special reason to use `Contains` here? like performance consideration?

Comment (smarterclayton) [inline pkg/kubectl/cmd/util/helpers.go]: It's probably more correct to use contains in these scenarios.

**Comment by smarterclayton** [comment]
Changes look to to me but will defer to TonyAdo for final review

**Comment by smarterclayton** [comment]
"Look ok"

File: pkg/kubectl/cmd/util/helpers.go (hunk 1 above)
Comment (bruceauyeung) [inline pkg/kubectl/cmd/util/helpers.go]: @AdoHe @smarterclayton its simpler and have better readability, as suggested by https://github.com/dominikh/go-simple

**Comment by AdoHe** [comment]
@bruceauyeung thanks.

**Comment by bruceauyeung** [comment]
@AdoHe
hi, this PR needs a `release-note-?` label to be merged, thanks !
cc @smarterclayton

**Comment by AdoHe** [comment]
@smarterclayton please help add `release-note-none` label

**Comment by smarterclayton** [comment]
@k8s-bot gce etcd3 e2e test this

**Comment by smarterclayton** [comment]
@k8s-bot gci gce e2e test this

**Comment by smarterclayton** [comment]
@k8s-bot cvm gce e2e test this

**Comment by AdoHe** [comment]
@k8s-bot kops aws e2e test this

**Comment by AdoHe** [comment]
@k8s-bot kops aws e2e test this

**Comment by bruceauyeung** [comment]
i just now did a `git rebase` operation to try to make sure e2e tests pass. would you plz label LGTM again ?
thanks !
@AdoHe @bruceauyeung @fabianofranz @smarterclayton

**Comment by AdoHe** [comment]
@k8s-bot gce etcd3 e2e test this

**Comment by AdoHe** [comment]
@bruceauyeung thanks.

**Comment by AdoHe** [comment]
@k8s-bot cri e2e test this

**Comment by AdoHe** [comment]
@k8s-bot cvm gce e2e test this

**Pull request merged**